### PR TITLE
Introduces .303 British as a calibre

### PR DESCRIPTION
**Pull request by DoctorBoomstick**
#### Summary
Content "adds .303 British as an ammo type in the game, plus corresponding FMJ, SP, and BTHP rounds, craft recipes, item groups, migrations, and the SMLE Mk III and No. 4 Mk I rifles"

#### Purpose of change
Whilst less common than other calibres, .303 is still a cartridge which is produced and shot for recreational and hunting purposes. Similarly, Enfield rifles are also commendably prolific collector guns. Apart for this and less importantly, I set a private challenge to myself to push out a First World War combat calibre with appropriate guns by November 11 at 11:11AM, given the historical circumstances and commemorations.

#### Describe the solution
I’ve added the .303 calibre as an in-game ammunition type. To this end, I’ve included boat-tailed hollo point, full metal jacketed boat-taled, and soft point rounds, plus their black powder and reloaded equivalents. Their crafting recipes have also been included, along with the Enfield Mk III and Enfield No. 4 Mk I battle rifles. All needed item groups have been included or otherwise sorted out, along with migrations and variants.

#### Describe alternatives you've considered

1.	Adding the 8mm (7.92x57mm) Mauser ammo to the game along side or instead of the venerable .303. I went for .303 ultimately as it’s overall more prolific to find, and more recognisable as a whole. My apologies K98 Mauser rifles and MG42 machine guns, y’all are just going to have to wait for another future PR.
2.	Adding more .303 guns as a whole, like the Ross Mk III rifles and the BREN and Lewis light machine guns. However, both in order to stick to my privately set 11:11 finish time and to keep the PR nice and bite sized, I decided to add the overall framework now and eventually more guns down the line in other, smaller PRs.
3.	Simply not adding to the sacrificial pyre of the gun bloat mountain we’ve already become infamous for. However, I think gun bloat’s fun. Some other people think gun bloat’s fun. Glory to the gun bloat.

#### Testing
All elements have been bug tested on the latest experimental builds, with the game loading just fine.
#### Additional context
What follows is a list of resources I consulted for information on modern production .303, as well as a few other miscellaneous links I utilised to determine other needed information.
https://www.hornady.com/ammunition/rifle/303-british-150-gr-sp?msclkid=f9d86fe3d11811ec91b5aaa8412d50ad#!/
https://topammomag.com/product/hornady-303-british-174gr-boattail-hollow-point-20-per-box/?msclkid=38fd0aa2d11a11eca23c11695209f1e8
https://www.ammoforsale.com/ammo-club/boat-tail-bullet/#:~:text=A%20boat%20tail%20bullet%20%28left%29%20next%20to%20a,it%20looks%20like%20the%20stern%20of%20a%20sailboat.?msclkid=22f4e7c1d11b11ecb25af5bcfb70fc76
[303 Cal .3105 174 gr FMJ BT - Hornady Manufacturing, Inc](https://www.hornady.com/bullets/rifle/303-cal-3105-174-gr-fmj-bt#!/)
[ENFIELD 303 NO1 MK III New and Used Price, Value, & Trends 2022 (truegunvalue.com)](https://truegunvalue.com/rifle/enfield-303-no1-mk-iii/price-historical-value/used/3)

### CODE REVIEW

File: data/json/items/ammo/303.json [hunk 1]
```diff
@@ -0,0 +1,110 @@
+[
+  {
+    "id": "303_bthp",
+    "type": "AMMO",
+    "name": { "str_sp": ".303 British BTHP" },
+    "description": "Modern manufactured .303 calibre cartridges, loaded with 174 GR boat-tailed hollo-point bullets.  Starting life as a black powder cartridge in service with British Long Lee rifles in the early nineteenth century, the cartridge evolved into a centre-fired smokeless-powder design and became the standard rifle and machinegun cartridge of the British commonwealth.  The round’s boat tail design gives it greater stability with lese aerial drag over long ranges, making it favourable for competition shooting, whilst the bullet’s hollow-point construction leaves behind larger wound channels when used in an offensive roll.",
```
Comment (TheShadowFerret) [inline data/json/items/ammo/303.json:6]:
```suggestion
    "description": "Modern manufactured .303 caliber cartridges, loaded with 174gr boat-tailed hollow-point bullets.  Starting life as a black powder cartridge in service with British Long Lee rifles in the early nineteenth century, the cartridge evolved into a center-fired smokeless-powder design and became the standard rifle and machinegun cartridge of the British commonwealth.  The round’s boat tail design gives it greater stability with lese aerial drag over long ranges, making it favorable for competition shooting, whilst the bullet’s hollow-point construction leaves behind larger wound channels when used in an offensive role.",
```

File: data/json/items/ammo/303.json [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    "relative": { "damage": { "range": 3, "damage_type": "bullet", "amount": 4, "armor_penetration": -6 } }
+  },
+  {
+    "id": "303_fmjbt",
+    "copy-from": "303_bthp",
+    "type": "AMMO",
+    "name": { "str_sp": ".303 British FMJBT" },
+    "description": "Modern manufactured full metal jacketed .303 calibre cartridges, loaded with 174 GR single-core boat tailed bullets.  Starting life as a black powder cartridge in service with British Long Lee rifles in the early nineteenth century, the cartridge evolved into a centre-fired smokeless-powder design and became the standard rifle and machinegun cartridge of the British commonwealth.  Featuring a boat tailed design, which allows the round to retain greater terminal performance and stability over longer ranges, the bullet possesses a solid metal jacket that grants it the capability to punch through thicker material at the cost of virtually non-existent expansion.",
```
Comment (TheShadowFerret) [inline data/json/items/ammo/303.json:38]:
```suggestion
    "description": "Modern manufactured full metal jacketed .303 caliber cartridges, loaded with 174gr single-core boat-tailed bullets.  Starting life as a black powder cartridge in service with British Long Lee rifles in the early nineteenth century, the cartridge evolved into a center-fired smokeless-powder design and became the standard rifle and machinegun cartridge of the British commonwealth.  Featuring a boat tailed design, which allows the round to retain greater terminal performance and stability over longer ranges, the bullet possesses a solid metal jacket that grants it the capability to punch through thicker material at the cost of virtually non-existent expansion.",
```

File: data/json/items/gun/303.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      {
+        "id": "smle_mk3",
+        "name": { "str": "SMLE No. 1 Mk III" },
+        "description": "A venerable English darling and icon of the First World War, the Short Magazine Lee-Enfield Mk III was the standard issue battle rifle of the British commonwealth from 1907.  Chambered for .303 ammunition and assigned with a sword style bayonet, the rifle makes use of 10-round detachable magazines.  With its full length wooden handguard, brass buttplate, and smooth action, the Lee-Enfield cuts a recognisable profile in either the trenches of the Great War or the zombie strewn wastelands.",
+        "weight": 1
+      },
+      {
+        "id": "number4_mki",
+        "name": { "str": "Lee-Enfield No. 4 Mk I" },
+        "description": "A Second World War progression of the original Lee-Enfield battle rifle adopted by the British Commonwealth; the No. 4 began its decades-long service life in 1939.  Externally similar to its forefather, the rifle possesses a reinforced receiver, adjusted nose cap, and heavier barrel.  Still chambered for the same .303 calibre cartridge, the No. 4 represents one of the most populous firearm imports in Us history."
```
Comment (TheShadowFerret) [inline data/json/items/gun/303.json:26]:
```suggestion
        "description": "A Second World War progression of the original Lee-Enfield battle rifle adopted by the British Commonwealth; the No. 4 began its decades-long service life in 1939.  Externally similar to its forefather, the rifle possesses a reinforced receiver, adjusted nose cap, and heavier barrel.  Still chambered for the same .303 caliber cartridge, the No. 4 represents one of the most populous firearm imports in U.S. history."
```

**Review by TheShadowFerret — COMMENTED** [review]
~~Who dares bring British spellings into America on a British weapon 😛~~

File: data/json/items/ammo/303.json [hunk 4]
```diff
@@ -0,0 +1,110 @@
+[
+  {
+    "id": "303_bthp",
+    "type": "AMMO",
+    "name": { "str_sp": ".303 British BTHP" },
+    "description": "Modern manufactured .303 caliber cartridges, loaded with 174gr boat-tailed hollow-point bullets.  Starting life as a black powder cartridge in service with British Long Lee rifles in the early nineteenth century, the cartridge evolved into a center-fired smokeless-powder design and became the standard rifle and machinegun cartridge of the British commonwealth.  The round’s boat tail design gives it greater stability with lese aerial drag over long ranges, making it favorable for competition shooting, whilst the bullet’s hollow-point construction leaves behind larger wound channels when used in an offensive role.",
+    "weight": "18 g",
```
Comment (nopenoperson) [inline data/json/items/ammo/303.json:7]: 18 + 7 (the case weight) matches up fairly well with data on https://www.sellier-bellot.cz/en/products/rifle-ammunition/list/1529/

File: data/json/items/ammo/303.json [hunk 5]
```diff
@@ -0,0 +1,110 @@
+[
+  {
+    "id": "303_bthp",
+    "type": "AMMO",
+    "name": { "str_sp": ".303 British BTHP" },
+    "description": "Modern manufactured .303 caliber cartridges, loaded with 174gr boat-tailed hollow-point bullets.  Starting life as a black powder cartridge in service with British Long Lee rifles in the early nineteenth century, the cartridge evolved into a center-fired smokeless-powder design and became the standard rifle and machinegun cartridge of the British commonwealth.  The round’s boat tail design gives it greater stability with lese aerial drag over long ranges, making it favorable for competition shooting, whilst the bullet’s hollow-point construction leaves behind larger wound channels when used in an offensive role.",
```
Comment (TheShadowFerret) [inline data/json/items/ammo/303.json:6]:
```suggestion
    "description": "Modern manufactured .303 caliber cartridges, loaded with 174gr boat-tailed hollow-point bullets.  Starting life as a black powder cartridge in service with British Long Lee rifles in the early nineteenth century, the cartridge evolved into a center-fired smokeless-powder design and became the standard rifle and machinegun cartridge of the British commonwealth.  The round’s boat tail design gives it greater stability with less aerial drag over long ranges, making it favorable for competition shooting, whilst the bullet’s hollow-point construction leaves behind larger wound channels when used in an offensive role.",
```

File: data/json/items/migration.json [hunk 6]
```diff
@@ -1153,6 +1153,12 @@
     "type": "MIGRATION",
     "replace": "m27_assault_rifle"
   },
+  {
+    "id": "number4_mki",
```
Comment (TheShadowFerret) [inline data/json/items/migration.json:1157]: Why is there a migration file in here if these are all new items 🤔

File: data/json/items/ammo/303.json (hunk 5 above)
Comment (nopenoperson) [reply data/json/items/ammo/303.json:6]: i think this is too wordy

**Comment by DoctorBoomstick** [comment]
@TheShadowFerret, I was under the belief that, in order for merged variants to work and spawn correctly, you had to create a migration entry for them as can be seen with stuff like the Auto 5 shotgun, the M27, etc. As things stand, the No. 4 Mk I is a variant under the Mk III.

**Comment by TheShadowFerret** [comment]
>I was under the belief that, in order for merged variants to work and spawn correctly, you had to create a migration entry for them as can be seen with stuff like the Auto 5 shotgun, the M27, etc. As things stand, the No. 4 Mk I is a variant under the Mk III.

Those other migrations are from old items being converted into variants of one item, new items don't need this. This migration references a phantom number4_mki, and can just be deleted.

File: data/json/itemgroups/Weapons_Mods_Ammo/ammo.json [hunk 7]
```diff
@@ -398,6 +404,9 @@
       { "item": "reloaded_762_51_incendiary", "prob": 5 },
       { "item": "reloaded_762_54R", "prob": 40 },
       { "item": "reloaded_762_m43", "prob": 20 },
+      { "item": "reloaded_303", "prob": 50 },
```
Comment (nopenoperson) [inline data/json/itemgroups/Weapons_Mods_Ammo/ammo.json:407]: .303 really should be  the rarest in this item group. it sees 0 use in modern military firearms, 303 rifles can be easily converted to fire 308/7.62x51 and export versions were frequently converted, surplus military ammo usually has berdan primers which complicates using the casings for reloading (even modern ammo sometimes has this). 

it should be in roughly the same itemgroups as 7.62x54r but a lot more rare since .303 doesnt get used in any modern military conflicts.

File: data/json/items/gun/303.json [hunk 8]
```diff
@@ -0,0 +1,57 @@
+[
+  {
+    "id": "smle_mk3",
+    "copy-from": "rifle_manual",
+    "looks_like": "ar15",
+    "type": "GUN",
+    "name": { "str": "Enfield battle rifle" },
+    "description": "An iconic weapon of both the First and Second World Wars, this bolt-action rifle has seen deployment in the hands of Commonwealth troops across Europe and beyond.  Outfitted with a 17-inch bayonet, firing the .303 calibre cartridge, and making use of 10-round Enfield magazines, the firearm’s reputation precedes it.",
+    "//": "Total unloaded weight of an Mk III is 3.96 kg, and a No. 4 is 4.11 kg.  total length of an Mk III is 1132mm, and a No. 4 is 1129mm.  Rounding to 4 kg of weight and 1130mm of length to meet in the middle.",
```
Comment (nopenoperson) [inline data/json/items/gun/303.json:9]: please do not use the variant system for this. you have enough data to make two separate items

 if someone wants do even more reasearch  or just make a small change, you have to delete a bunch of stuff in the entry file, change a line in every itemgroup where the gun is used, add migration entries and probably some extra linting instead of simply changing a few numbers in a single json file. 

https://github.com/CleverRaven/Cataclysm-DDA/pull/61895/files here i had to change stuff in 5 different documents just to add an adjustable folding stock to the scar_l

File: data/json/items/gun/303.json [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+    "bashing": 12,
+    "material": [ "steel", "wood" ],
+    "color": "brown",
+    "ammo": [ "303" ],
+    "ranged_damage": { "damage_type": "bullet", "amount": 9 },
+    "dispersion": 100,
+    "durability": 8,
+    "blackpowder_tolerance": 60,
+    "barrel_volume": "250 ml",
+    "default_mods": [ "sword_bayonet" ],
```
[the remark this answers is not in the document]
Comment (DoctorBoomstick) [reply data/json/items/gun/303.json:39]: The issue of the sword bayonet is honestly an interesting one, cause I don’t think that you can define a probability for it’s spawn within the nested item group. Looking at stuff like the mod attachments for the M38 DMR, it looks like that they’d always spawn anyway, which means that defining the bennet in a nested item group as apposed to a default mod would have no functional difference, save for the amount of JSON present. That being said, theoretically I could define a nested item group and spawn for  a lee-Enfield without the attachment, and another, rarer spawn for an Enfield with a bayonet, but then this would honestly be more lines of JSON than needed, and would bring into question all the other weapons that also spawn with a bayonet by default, like the 1896 trench gun.
That being said, splitting the No. 4 and Mk III into two items, the former with the spike bayonet depicted here, is a decent idea. The only change I’d suggest of doing for the spike is giving it piercing damage rather than cutting.

[the remark this answers is not in the document]
Comment (nopenoperson) [reply data/json/items/gun/303.json:82]: the "STAB" flag gives it piercing damage

[the remark this answers is not in the document]
Comment (DoctorBoomstick) [reply data/json/items/gun/303.json:82]: I see. I'll get around to adding it when I've got more time to work on this Pr, but thanks for the info and the framework for what the attachment should look like.

**Comment by bombasticSlacks** [comment]
This looks good to me, could you add the barrel lengths to the guns? #64961 gives an example.

**Comment by DoctorBoomstick** [comment]
> This looks good to me, could you add the barrel lengths to the guns? #64961 gives an example.

Certainly. Do I need to just add, "barrel_length": "assorted amount", in the gun JSON, or do I also need to fiddle around with the ammo? Not totally sure how this works. Also, side note, but on the matter of gun mods, does this barrel length business also work with mods not specifically assigned to the barrel slot of a weapon? Asking cause in #64934, I’m adding conversion kits which also can change their weapons’ barrel length, and those fit into the bore slot, and I’m wondering if this’ll be an issue.

**Comment by bombasticSlacks** [comment]
> Certainly. Do I need to just add, "barrel_length": "assorted amount", in the gun JSON, or do I also need to fiddle around with the ammo? Not totally sure how this works. Also, side note, but on the matter of gun mods, does this barrel length business also work with mods not specifically assigned to the barrel slot of a weapon? Asking cause in https://github.com/CleverRaven/Cataclysm-DDA/pull/64934, I’m adding conversion kits which also can change their weapons’ barrel length, and those fit into the bore slot, and I’m wondering if this’ll be an issue.

was gonna talk with you about the conversion kits at some point.

for this PR just add barrel length, I'll comment on the conversion kit PR when I get a chance. Sorry been juggling a lot of project :)

**Comment by bombasticSlacks** [comment]
extra crafts giving you conflicts, sorry :(

**Comment by DoctorBoomstick** [comment]
> extra crafts giving you conflicts, sorry :(
Afraid I’m not spotting the issue. The ammo crafts look ok, and I can’t see anything that’s amiss, unless I accidentally overwrote some changes that I’m not aware of.

**Comment by DoctorBoomstick** [comment]
> > extra crafts giving you conflicts, sorry :(
> > Afraid I’m not spotting the issue. The ammo crafts look ok, and I can’t see anything that’s amiss, unless I accidentally overwrote some changes that I’m not aware of.

Yes. I’m a dumb dumb. Just figured out what the mistake was. Please disregard this.

**Comment by DoctorBoomstick** [comment]
Looks like the failing tests are a result of something having to do with monster json, which I didn’t touch for this PR.

**Comment by bombasticSlacks** [comment]
are you okay with me rebasing this to try to fix the errors?

**Comment by DoctorBoomstick** [comment]
> are you okay with me rebasing this to try to fix the errors.

Perfectly fine with that, cause for the life of me, I can’t tell what’s the issue. Apologies for being completely illiterate where GGIT is concerned and pestering you about this.

**Comment by bombasticSlacks** [comment]
barrel stuff looks good 😄
